integration/basic_test.go

```diff
@@ -367,6 +367,33 @@ func TestCollectionName(t *testing.T) {
 			},
 			alt: "Invalid collection name: 'testcollectionname.'",
 		},
+		"Null": {
+			collection: "\x00",
+			err: &mongo.CommandError{
+				Name:    "InvalidNamespace",
+				Code:    73,
+				Message: "namespaces cannot have embedded null characters",
+			},
+			alt: "Invalid collection name: 'testcollectionname.\x00'",
+		},
+		"Dot": {
+			collection: "collection.name",
+		},
+		"Space": {
+			collection: " ",
+		},
+		"NonLatin": {
+			collection: "coleção",
```

Comment:
Those still look quite Latin to me, like extended Latin. Let's maybe add something Japanese, or Greek, or Cyrillic to make it look more non-latin (just to be absolutely sure how it works)?

Reply:
Sure, let's try japanese because I know how to write collection :hugs: